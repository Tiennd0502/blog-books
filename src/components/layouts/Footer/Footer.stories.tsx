import React from 'react';
import Storybook from '@storybook/react';

import Footer from './index';

import '@styles/main.css';

export default {
  title: 'Components/Footer',
  component: Footer,
} as Storybook.ComponentMeta<typeof Footer>;

const Template: Storybook.ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
);

export const Default = Template.bind({});
