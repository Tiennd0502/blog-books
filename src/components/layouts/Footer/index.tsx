import React from 'react';
import { Box, Text, Container } from '@chakra-ui/react';

const Footer: React.FC = () => {
  return (
    <Box as="footer" bg="blue.700" data-testid="footer" color="light">
      <Container padding="14px 14px 22px" textAlign="center">
        <Text>
          OpenSource Textbook empowers and connects authors and readers through
          open access to a global reading and publishing platform.
        </Text>
        <Text>Vancouver Community College, Vancouver, B.C., Canada</Text>
        <Text>&copy; 2022 by NIKEA</Text>
      </Container>
    </Box>
  );
};

export default Footer;
