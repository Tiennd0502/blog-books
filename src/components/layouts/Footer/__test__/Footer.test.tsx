import { render } from '@testing-library/react';

import Footer from '../index';

describe('Footer component', () => {
  test('Should render Footer component', () => {
    const { getByTestId } = render(<Footer />);
    const image = getByTestId('footer') as HTMLImageElement;

    expect(image).toBeInTheDocument();
  });

  test('Matches snapshot', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
