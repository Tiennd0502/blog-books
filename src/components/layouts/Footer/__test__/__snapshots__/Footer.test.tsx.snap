// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Matches snapshot 1`] = `
<div>
  <footer
    class="css-1ry9sjz"
    data-testid="footer"
  >
    <div
      class="chakra-container css-caqc3d"
    >
      <p
        class="chakra-text css-0"
      >
        OpenSource Textbook empowers and connects authors and readers through open access to a global reading and publishing platform.
      </p>
      <p
        class="chakra-text css-0"
      >
        Vancouver Community College, Vancouver, B.C., Canada
      </p>
      <p
        class="chakra-text css-0"
      >
        Â© 2022 by NIKEA
      </p>
    </div>
  </footer>
</div>
`;
